lms/views/basic_lti_launch.py

```diff
@@ -137,12 +137,36 @@ def blackboard_copied_basic_lti_launch(self):
         For more about Blackboard course copy and the resource_link_id_history
         param see the BlackboardCopied predicate's docstring.
         """
-        # The ``blackboard_copied=True`` view predicate ensures that this view
-        # won't be called if there isn't a resource_link_id_history param and a
-        # matching document_url in the DB.
+        return self.course_copied_basic_lti_launch("resource_link_id_history")
+
+    @view_config(brightspace_copied=True)
+    def brightspace_copied_basic_lti_launch(self):
+        """
+        Respond to a launch of a newly-copied Brightspace assignment.
+
+        Find the document_url for ext_d2l_resource_link_id_history and make a
+        copy of it with the new resource_link_id, then lanuch the assignment as
+        normal.
+
+        For more about Brightspace course copy and the
+        ext_d2l_resource_link_id_history param see the BrightspaceCopied
+        predicate's docstring.
+        """
+        return self.course_copied_basic_lti_launch("ext_d2l_resource_link_id_history")
+
+    def course_copied_basic_lti_launch(self, param_name):
+        """
+        Respond to a launch of a newly-copied assignment.
+
+        Helper method for the *_copied_basic_lti_launch() methods above.
+
+        :param param_name: the name of the launch param that contains the
+        resource_link_id of the original assignment that this assignment was
+        copied from
+        """
```

Comment:
Added a helper method for `blackboard_copied_basic_lti_launch()` and `brightspace_copied_basic_lti_launch()`, since the two methods are exactly the same except for the `param_name`. I prefer having two separate views that call a helper method, rather than a single view that looks for both param names, the code works out simpler.